src/main/java/com/aws/greengrass/disk/spool/DiskSpoolDAO.java

```diff
@@ -68,159 +87,189 @@ public DiskSpoolDAO(NucleusPaths paths) throws IOException {
         url = String.format(DATABASE_CONNECTION_URL, path);
     }
 
+    /**
+     * Initialize the database connection.
+     *
+     * @throws SQLException if db is unable to be created
+     */
+    public void initialize() throws SQLException {
+        try (LockScope ls = LockScope.lock(connectionLock.writeLock())) {
+            close();
+            logger.atDebug().kv("url", url).log("Creating DB connection");
+            connection = getDbInstance();
+        }
+    }
+
+    /**
+     * Close DAO resources.
+     */
+    public void close() {
+        try (LockScope ls = LockScope.lock(connectionLock.readLock())) {
```

Comment:
why only a readlock? We're closing the connection which would harm anyone else using this connection

Reply:
whoops that was a miss, fixing

Reply:
fixed